libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -451,3 +451,17 @@ bool AP_GPS_SBF::prepare_for_arming(void) {
 
     return is_logging;
 }
+
+void AP_GPS_SBF::inject_data(const uint8_t *data, uint16_t len)
+{
+    // Septentrio configuration step fails if RTCM corrections are injected while configuring
+    // so wait until configuration is completed, before injecting data
+    if (state.status != AP_GPS::NO_GPS) {
```

Comment:
This should just be:
```
if (is_configured()) {
    GPS_Backend::inject_data(data, len);
} else {
    Debug("Refusing to inject data until the GPS is configured");
}
```

Reply:
That will flood the debug output a bit... but yes it avoids code duplication.
Why use is_configured instead of status != NO_GPS ?

Reply:
Debug output is only done if you set a #define, it doesn't normally show up on the debug console, it's also already very flood like, so I think that's fine.

`is_configured()` has a couple of merits:
1. Deconflicts any chance for injected data to confuse the configuration process (IE if you inject something other then RTCM that get's ACK'd there is a chance of the configuration portion seeing this message and thinking it was NACK'd)
1. Allows the uploading/sending of custom commands much earlier then waiting for GPS lock (for example if you are indoors). (It also provides positive assurance the GPS has actually booted all the way, which resolves your other issues quite handily).
1. `NO_GPS` is pointless. If the driver can be called, it will always be at least `NO_FIX` which means that your desired patch is a no-op. But waiting on a fix > `NO_FIX` will be painful for indoor operations/testing.